### PR TITLE
librdmacm: Remove unneeded condition

### PR DESCRIPTION
**Pull request by yuvalshaia**
The check makes no difference - remove it.

Signed-off-by: Yuval Shaia <yuval.shaia@oracle.com>

### CODE REVIEW

**Comment by rleon** [comment]
@jgunthorpe Can it be that "=" was written instead of "=="?

Thanks

**Comment by jgunthorpe** [comment]
@rleon Looks like the assignment is necessary to me..

**Comment by rleon** [comment]
@jgunthorpe Thanks, I looked into the code now and cmd.src_size is not initialized, so yes assignment is needed.

File: librdmacm/cma.c [hunk 1]
```diff
@@ -880,8 +880,8 @@ static int rdma_resolve_addr2(struct rdma_cm_id *id, struct sockaddr *src_addr,
 	CMA_INIT_CMD(&cmd, sizeof cmd, RESOLVE_ADDR);
 	id_priv = container_of(id, struct cma_id_private, id);
 	cmd.id = id_priv->handle;
-	if ((cmd.src_size = src_len))
-		memcpy(&cmd.src_addr, src_addr, src_len);
+	cmd.src_size = src_len;
```
Comment (gladkovdmitry17) [inline librdmacm/cma.c]: @yuvalshaia @rleon @jgunthorpe Just out of curiosity - why this change is needed?
It assigns to the `ucma_abi_resolve_addr` command a source size of an address and if this size is 0, it won't do the an excessive call of the `memset` function.
The `cmd->src_addr` is 0, because the structure is zeromemd in the `CMA_INIT_CMD`.

Comment (rleon) [reply librdmacm/cma.c]: @gladkovdmitry17 IMHO, it makes code cleaner.

Comment (yuvalshaia) [reply librdmacm/cma.c]: Yeah, the existing code is unusual, just see how len is not checked in dest_addr

**Comment by yuvalshaia** [comment]
Closing this pull request and submitting new one with a revised commit message

File: librdmacm/cma.c (hunk 1 above)
Comment (gladkovdmitry17) [reply librdmacm/cma.c]: @rleon @yuvalshaia Ok, it was just out of curiosity.
Yes, it will be similar to `dest_addr`
